### PR TITLE
feat(websh): add sudo MFA support via event WebSocket

### PR DESCRIPTION
**Pull request by eunyoung14**
## Summary

Add sudo MFA support to the CLI's Websh terminal, matching the Alpacon web terminal behavior. When a user types `sudo` and MFA is required, the CLI automatically opens the browser for verification and approves the request after completion.

## Changes

- **`api/event/subscribe.go`** (new): `CreateEventSession()` and `SubscribeSudoEvent()`—event WebSocket session and sudo subscription management
- **`api/event/sudolistener.go`** (new): `SudoListener`—goroutine-based event WebSocket client with auto-reconnect (exponential backoff 1s→30s), MFA event detection, browser auto-open, MFA completion polling, and self-approve API call. Graceful shutdown via `Stop()` which closes the WebSocket to unblock `ReadMessage`
- **`api/event/subscribe_test.go`** (new): 4 tests (session creation, subscription, server errors)
- **`api/event/sudolistener_test.go`** (new): 7 tests (MFA parsing, non-MFA ignore, Stop idempotent, Stop closes connection, WebSocket connect/listen, polling timeout)
- **`cmd/websh/websh.go`**: `setupSudoListener()` wires event subscription before `OpenNewTerminal()`. Best-effort—older servers without events API are silently skipped

## Flow

```
user types sudo → server detects → sends mfa_request via event WebSocket
→ CLI prints "Sudo MFA required" + opens browser
→ CLI polls MFA completion (2s interval, 60s timeout)
→ user completes MFA in browser
→ CLI calls POST /api/approvals/sudo-policies/self-approve/
→ server approves → sudo executes
```

### Universal
- [x] Code follows project style guidelines
- [x] Self-reviewed the code
- [x] No hardcoded secrets or credentials

### Golang
- [x] Error handling is complete
- [x] No goroutine leaks (Stop closes WebSocket + done channel)
- [x] Tests include edge cases
- [x] `go vet` passes clean

## Test plan

- [x] Unit tests: 11 new tests covering event session, subscription, listener lifecycle, MFA parsing, shutdown
- [ ] Manual: `alpacon websh root@server` → `sudo <command>` → verify browser opens, MFA completes, command executes
- [ ] Manual: verify graceful degradation on server without events API

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: cmd/websh/websh.go [hunk 1]
```diff
@@ -197,6 +204,31 @@ func extractValue(args []string, i int) (string, int) {
...
+		if !isNotFoundError(err) {
+			utils.CliWarning("Sudo MFA listener unavailable: %s\n", err)
+		}
+		return nil
+	}
+
+	if err := event.SubscribeSudoEvent(ac, eventSession.ChannelID, sessionID); err != nil {
+		if !isNotFoundError(err) {
+			utils.CliWarning("Sudo MFA listener unavailable: %s\n", err)
+		}
```
[the remark this answers is not in the document]
Comment (eunyoung14) [reply cmd/websh/websh.go]: Fixed. Removed trailing \n from both CliWarning format strings.

File: api/event/sudolistener.go [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+	approvalID := event.Payload.ApprovalRequestID
+
+	if mfaURL == "" || approvalID == "" {
+		return
+	}
+
+	// Print MFA prompt to stderr (terminal is in raw mode, use \r\n)
+	fmt.Fprintf(os.Stderr, "\r\n\033[33mSudo MFA required. Opening browser for verification...\033[0m\r\n")
+	utils.OpenBrowser(mfaURL)
+	fmt.Fprintf(os.Stderr, "  %s\r\n", mfaURL)
```
[the remark this answers is not in the document]
Comment (eunyoung14) [reply api/event/sudolistener.go]: Fixed. Added `<-sl.done` check before and after `mfaMu.Lock()` so shutdown prevents browser opens and API calls.

File: api/event/sudolistener_test.go [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+			return
+		}
+		defer func() { _ = conn.Close() }()
+
+		// Send a non-MFA message
+		msg := `{"payload":{"type":"info","query":"status"}}`
+		_ = conn.WriteMessage(websocket.TextMessage, []byte(msg))
+
+		// Wait for client to read it before closing
+		time.Sleep(50 * time.Millisecond)
```
[the remark this answers is not in the document]
Comment (eunyoung14) [reply api/event/sudolistener_test.go]: Fixed. Same as C4 — test now uses sync channel instead of sleep.

File: cmd/websh/websh_test.go [hunk 4]
```diff
@@ -258,6 +259,51 @@ func TestCommandParsing(t *testing.T) {
...
+		},
+		{
+			name:     "case insensitive NOT FOUND",
+			err:      fmt.Errorf("NOT FOUND"),
+			expected: true,
+		},
+		{
+			name:     "wrapped error is not matched (strict)",
+			err:      fmt.Errorf("failed to create event session: Not found"),
+			expected: false,
```
[the remark this answers is not in the document]
Comment (eunyoung14) [reply cmd/websh/websh_test.go]: Fixed. isNotFoundError now matches wrapped errors via HasSuffix. Test updated accordingly.
